### ISSUE TITLE
Avoiding vec allocation when surpressing whitespace splitting

### ISSUE THREAD

**Issue by casey** [soenkehahn/stir#22]
Is it possible to make slices have the same behavior as vectors, so that you don't need to allocate to avoid splitting? Arrays would also be nice, to avoid the `&`.

I.e:

```rust
cmd!("cat", ["file name with spaces.txt"])
```

This could also be done at the macro level, with the `cmd!` macro treating bracket-wrapped arguments (`[…]`) specially.

**Comment by soenkehahn** [comment]
I created this PR for supporting slices: #27. However, under the hood, `cmd!` collects arguments into a `Vec<String>`. So there *is* going to be allocations. There's no reason to disallow slices, so I feel inclined to merge #27. But yeah, it doesn't really solve this ticket, at least not in spirit. What do you think? Do you want to avoid allocations just for performance?

For arrays: Is this only for avoiding the `&`? Or is there another advantage in supporting arrays?

**Comment by casey** [comment]
> I created this PR for supporting slices: #27. However, under the hood, `cmd!` collects arguments into a `Vec<String>`. So there _is_ going to be allocations. There's no reason to disallow slices, so I feel inclined to merge #27. But yeah, it doesn't really solve this ticket, at least not in spirit. What do you think? Do you want to avoid allocations just for performance?

I think it solves this ticket. With `vec![…]` there would be an extra allocation for the vector, which #27 fixes. The contents are still allocated, but I wasn't thinking about that.

> For arrays: Is this only for avoiding the `&`? Or is there another advantage in supporting arrays?

Yeah, just avoiding the `&`. Look at how clean it is:

```rust
cmd!("cat", ["file name with spaces.txt"])
``

**Comment by casey** [comment]
I think using a macro to generate different array impls is probably not necessary. Lots of people are on 1.51+, and it's only a convenience feature anyways.

**Comment by soenkehahn** [comment]
Merged #27.
